### PR TITLE
Grouping z-index variables

### PR DESCRIPTION
**Pull request by luixxiul**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

Based on #1956 to avoid a case like #1910. If OK I will squash them and push again.

main components: x000s
sub components: x00s

### CODE REVIEW

File: less/about/preferences.less [hunk 1]
```diff
@@ -11,7 +11,6 @@
   position: fixed;
   width: @sideBarWidth;
   height: 100%;
-  z-index: 100;
```
Comment (luixxiul) [inline less/about/preferences.less]: This one is no longer needed as ``.helpfulHints`` was completely rewritten

File: less/about/preferences.less [hunk 2]
```diff
@@ -130,7 +129,6 @@
     left: ~"calc(100% - 6px)";
     transform: rotateZ(-90deg);
     position: absolute;
-    z-index: 400;
```
Comment (luixxiul) [inline less/about/preferences.less]: ditto

File: less/window.less [hunk 3]
```diff
@@ -6,9 +6,13 @@
 
 :root {
   --default-font-family: "Arial";
+  --zindex-frameWrapper: @zindexWindow;
+  --zindex-notActive: calc(@zindexWindow - 100);
+  --zindex-isPreview: calc(@zindexWindow + 100);
+  --zindex-fullScreen: @zindexFullScreen;
+  --zindex-fullScreenBanner: calc(@zindexFullScreen + 100);
```
Comment (bbondy) [inline less/window.less]: These aren't needed here. You only need this for things you want to access in JS as well.

File: less/window.less [hunk 4]
```diff
@@ -60,7 +64,7 @@ html,
 .frameWrapper {
   height: 100%;
   width: 100%;
-  z-index: 2;
+  z-index: var(--zindex-frameWrapper);
```
Comment (bbondy) [inline less/window.less]: You can use the less variable here directly.

Comment (bbondy) [inline less/window.less]: via z-index:  @zindexWindow

File: less/tabs.less [hunk 5]
```diff
@@ -4,6 +4,12 @@
 
 @import "variables.less";
 
+:root {
+  --zindex-tabs: @zindexTabs;
+  --zindex-thumbnail: calc(@zindexTabs + 100);
+  --zindex-dragIndicator: calc(@zindexTabs + 100);
+}
```
Comment (bbondy) [inline less/tabs.less]: Remove CSS variables

**Comment by bbondy** [comment]
Looks good but I think the CSS variables are adding unneeded complexity. Please just use the less variables directly. Thanks!

**Comment by luixxiul** [comment]
I think this is ready.

**Comment by bbondy** [comment]
++

**Pull request merged**